### PR TITLE
Upgrade che with cli

### PR DESCRIPTION
**Pull request by themr0c**
* Refactoring installation procedures on OpenShift 3.
* Adding upgrade procedures using the CLI on OpenShift 3 and 4.

### CODE REVIEW

File: src/main/pages/che-7/installation-guide/assembly_updating-che.adoc [hunk 1]
```diff
@@ -15,8 +15,24 @@ summary:
...
+
+Following scenarios are covered:
+
+* xref:upgrading-from-{prod-id-short}-{prod-prev-ver}-to-{prod-id-short}-{prod-ver}-on-openshift-4-using-the-operator_{context}[]
+
+* xref:upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-4-using-the-cli-management-tool_{context}[]
+
+* xref:upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool_{context}[]
+
+WARNING: The method that was used to install the {prod-short} {prod-prev-ver} instance determines the method to proceed to the upgrade.
```
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/assembly_updating-che.adoc]:
```suggestion
WARNING: The method that was used to install the {prod-short} {prod-prev-ver} instance determines the method to proceed with for the upgrade.
```

File: src/main/pages/che-7/installation-guide/assembly_updating-che.adoc [hunk 2]
```diff
@@ -15,8 +15,24 @@ summary:
 
 :context: updating-che
 
+This chapter describes how to upgrade from {prod-short} {prod-prev-ver} instance to {prod-short} {prod-ver}.
+
+Following scenarios are covered:
+
+* xref:upgrading-from-{prod-id-short}-{prod-prev-ver}-to-{prod-id-short}-{prod-ver}-on-openshift-4-using-the-operator_{context}[]
+
+* xref:upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-4-using-the-cli-management-tool_{context}[]
```
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/assembly_updating-che.adoc]: The `prod-short` and `prod-ver` attribute names shouldn't be in the IDs.

File: src/main/pages/che-7/installation-guide/assembly_updating-che.adoc [hunk 3]
```diff
@@ -15,8 +15,24 @@ summary:
...
+
+* xref:upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool_{context}[]
+
+WARNING: The method that was used to install the {prod-short} {prod-prev-ver} instance determines the method to proceed to the upgrade.
+
 include::assembly_upgrading-from-che-6-to-che-7-on-openshift-4-using-the-operator.adoc[leveloffset=+1]
 
+include::proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-4-using-the-cli-management-tool.adoc[leveloffset=+1]
```
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/assembly_updating-che.adoc]: The prod-short and prod-ver attribute names shouldn't be in the file names.

File: src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool.adoc [hunk 4]
```diff
@@ -0,0 +1,30 @@
+[id="upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool_{context}"]
```
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool.adoc]: The prod-short and prod-ver attribute names shouldn't be in the IDs.

File: src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool.adoc [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+= Upgrading from {prod-short} {prod-prev-ver} to {prod-short} {prod-ver} on OpenShift 3 using the CLI management tool
+
+This section describes how to upgrade from {prod-short} {prod-prev-ver} to {prod-short} {prod-ver} on OpenShift 3 using the CLI management tool.
+
+.Prerequisites
+
+* And administrative account on an OpenShift 3 instance.
+* A running instance of {prod} runninng on OpenShift 3.
```
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool.adoc]:
```suggestion
* A running instance of {prod} running on OpenShift 3.
```

File: src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool.adoc [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+* This instance was installed using the CLI management tool.
+* `chectl` management tool is installed. See link:{site-baseurl}che-7/installing-the-chectl-management-tool/[Installing the `chectl` management tool].
```
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool.adoc]:
```suggestion
* The `chectl` management tool is installed. See link:{site-baseurl}che-7/installing-the-chectl-management-tool/[Installing the `chectl` management tool].
```

File: src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool.adoc [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+.Procedure
+
+. In all running workspaces in the {prod-short} {prod-prev-ver} instance, save and push changes back to the git repositories.
```
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool.adoc]:
```suggestion
. In all running workspaces in the {prod-short} {prod-prev-ver} instance, save and push changes back to the Git repositories.
```

File: src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool.adoc [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+. Shut down all workspaces in the {prod-short} {prod-prev-ver} instance.
+
+. Run following command:
```
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool.adoc]:
```suggestion
. Run the following command:
```

File: src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool.adoc [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
++
+[subs="+attributes,+quotes"]
+----
+chectl server:update
```
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool.adoc]:
```suggestion
$ chectl server:update
```

File: src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool.adoc [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+----
+
+.Verification steps
+
+. Login into the {prod-short} instance.
```
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-3-using-the-cli-management-tool.adoc]:
```suggestion
. Log in to the {prod-short} instance.
```

File: src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-4-using-the-cli-management-tool.adoc [hunk 11]
```diff
@@ -0,0 +1,30 @@
+[id="upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-4-using-the-cli-management-tool_{context}"]
+= Upgrading from {prod-short} {prod-prev-ver} to {prod-short} {prod-ver} on OpenShift 4 using the CLI management tool
+
+This section describes how to upgrade from {prod-short} {prod-prev-ver} to {prod-short} {prod-ver} on OpenShift 4 using the CLI management tool.
+
+.Prerequisites
+
+* And administrative account on an OpenShift 4 instance.
+* A running instance of {prod} runninng on OpenShift 4.
```
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-4-using-the-cli-management-tool.adoc]:
```suggestion
* A running instance of {prod} running on OpenShift 4.
```

File: src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-4-using-the-cli-management-tool.adoc [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+* This instance was installed using the CLI management tool, not using OperatorHub.
+* `chectl` management tool is installed. See link:{site-baseurl}che-7/installing-the-chectl-management-tool/[Installing the `chectl` management tool].
```
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-4-using-the-cli-management-tool.adoc]:
```suggestion
* The `chectl` management tool is installed. See link:{site-baseurl}che-7/installing-the-chectl-management-tool/[Installing the `chectl` management tool].
```

File: src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-4-using-the-cli-management-tool.adoc [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+
+.Procedure
+
+. In all running workspaces in the {prod-short} {prod-prev-ver} instance, save and push changes back to the git repositories.
```
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-4-using-the-cli-management-tool.adoc]:
```suggestion
. In all running workspaces in the {prod-short} {prod-prev-ver} instance, save and push changes back to the Git repositories.
```

File: src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-4-using-the-cli-management-tool.adoc [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+
+. Shut down all workspaces in the {prod-short} {prod-prev-ver} instance.
+
+. Run following command:
++
+[subs="+attributes,+quotes"]
+----
+chectl server:update
```
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-4-using-the-cli-management-tool.adoc]:
```suggestion
$ chectl server:update
```

File: src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-4-using-the-cli-management-tool.adoc [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+----
+
+.Verification steps
+
+. Login into the {prod-short} instance.
```
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/proc_upgrading-from-prod-short-prod-prev-ver-to-prod-short-prod-ver-on-openshift-4-using-the-cli-management-tool.adoc]:
```suggestion
. Log in to the {prod-short} instance.
```

File: src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator-and-ssl.adoc [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+. Run the following command to create the {prod-short} instance:
++
+[subs="+quotes,+attributes",options="nowrap"]
+----
+$ chectl server:start --platform=openshift --installer=operator --domain=<OPENSHIFT_APPS_URL> --tls
+----
+
+.Verification steps
+
+. The previous command output end with:
```
Comment (rkratky) [inline src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator-and-ssl.adoc]:
```suggestion
. The output of the previous command ends with:
```

File: src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator-and-ssl.adoc [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
++ 
+----
+Command server:start has completed successfully.
+----
+
+. Navigate to the {prod-short} cluster instance. The domain is now prefixed with HTTPS and using _Let’s Encrypt_ ACME certificates: `https:/che-che.<OPENSHIFT_APPS_URL>`.
```
Comment (rkratky) [inline src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator-and-ssl.adoc]:
```suggestion
. Navigate to the {prod-short} cluster instance. The domain is now prefixed with HTTPS and using _Let’s Encrypt_ ACME certificates: `++https://++che-che.__<OPENSHIFT_APPS_URL>__`.
```

File: src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator-and-ssl.adoc [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+[id="installing-{prod-id-short}-on-openshift-3-using-the-operator-and-ssl_{context}"]
+= Installing {prod-short} on OpenShift 3 using the Operator and SSL
+
+This section describes how to install {prod-short} on OpenShift 3 with the CLI management tool, using the Operator method and the SSL option.
+
+.Prerequisites
+
+* A running instance of OpenShift 3.11.
+* Administrator rights on this OpenShift 3 instance.
+* The `oc` OpenShift 3.11 CLI management tool is installed and configured. See link:https://access.redhat.com/documentation/en-us/openshift_container_platform/3.11/html/cli_reference/cli-reference-get-started-cli#installing-the-cli[Installing the OpenShift 3.11 CLI].
```
Comment (rkratky) [inline src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator-and-ssl.adoc]: I think we should link to docs.openshift.com

File: src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator.adoc [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+[id="installing-{prod-id-short}-on-openshift-3-using-the-operator_{context}"]
+= Installing {prod-short} on OpenShift 3 using the Operator
+
+This section describes how to install {prod-short} on OpenShift 3 with the CLI management tool, using the Operator method.
+
+.Prerequisites
+
+* A running instance of OpenShift 3.11.
+* Administrator rights on this OpenShift 3 instance.
+* The `oc` OpenShift 3.11 CLI management tool is installed and configured. See link:https://access.redhat.com/documentation/en-us/openshift_container_platform/3.11/html/cli_reference/cli-reference-get-started-cli#installing-the-cli[Installing the OpenShift 3.11 CLI].
```
Comment (rkratky) [inline src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator.adoc]: I think we should link to docs.openshift.com

File: src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator.adoc [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+[subs="+quotes,+attributes"]
+----
+$ oc new-project che
+----
+
+. Run the following command to create the {prod-short} instance:
++
+[subs="+quotes,+attributes",options="nowrap"]
+----
+$ chectl server:start --platform=openshift --installer=operator --domain=<OPENSHIFT_APPS_URL>
```
Comment (rkratky) [inline src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator.adoc]:
```suggestion
$ chectl server:start --platform=openshift --installer=operator \
  --domain=__<OPENSHIFT_APPS_URL>__
```

File: src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator.adoc [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+----
+
+.Verification steps
+
+. The previous command output end with:
```
Comment (rkratky) [inline src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator.adoc]:
```suggestion
. The output of the previous command ends with:
```

File: src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator.adoc [hunk 22]
```diff
@@ -0,0 +1,42 @@
...
++ 
+----
+Command server:start has completed successfully.
+----
+
+. Navigate to the {prod-short} cluster instance: `http:/che-che.<OPENSHIFT_APPS_URL>`.
```
Comment (rkratky) [inline src/main/pages/che-7/overview/proc_installing-che-on-openshift-3-using-the-operator.adoc]:
```suggestion
. Navigate to the {prod-short} cluster instance: `++http://++che-che.__<OPENSHIFT_APPS_URL>__`.
```

File: src/main/pages/che-7/overview/proc_preparing-openshift-3-for-installing-che.adoc [hunk 23]
```diff
@@ -0,0 +1,49 @@
+// installing-che-on-openshift-3-using-the-operator
+
+[id="preparing-openshift-3-for-installing-che_{context}"]
+= Preparing OpenShift 3 for installing {prod-short}
+
+.Prerequisites
+
+* A running instance of OpenShift 3.11.
+* Administrator rights on this OpenShift 3 instance.
+* The `oc` OpenShift 3.11 CLI management tool is installed and configured. See link:https://access.redhat.com/documentation/en-us/openshift_container_platform/3.11/html/cli_reference/cli-reference-get-started-cli#installing-the-cli[Installing the OpenShift 3.11 CLI].
```
Comment (rkratky) [inline src/main/pages/che-7/overview/proc_preparing-openshift-3-for-installing-che.adoc]: I think we should link to docs.openshift.com
```suggestion
* The `oc` OpenShift 3.11 CLI management tool is installed and configured. See link:https://docs.openshift.com/container-platform/3.11/cli_reference/get_started_cli.html#installing-the-cli[Installing the OpenShift 3.11 CLI].
```

File: src/main/pages/che-7/overview/proc_preparing-openshift-3-for-installing-che.adoc [hunk 24]
```diff
@@ -0,0 +1,49 @@
...
+.Procedure
+
+. Log in to OpenShift. See link:https://access.redhat.com/documentation/en-us/openshift_container_platform/3.11/html/cli_reference/cli-reference-get-started-cli#basic-setup-and-login[Basic Setup and Login].
++
+[subs="+attributes,+quotes",options="nowrap"]
+----
+$ oc login
+----
+
+. Run following command to verify the version of the `oc` OpenShift CLI management tool is 3.11.
```
Comment (rkratky) [inline src/main/pages/che-7/overview/proc_preparing-openshift-3-for-installing-che.adoc]:
```suggestion
. Run the following command to verify that the version of the `oc` OpenShift CLI management tool is 3.11:
```

File: src/main/pages/che-7/overview/proc_preparing-openshift-3-for-installing-che.adoc [hunk 25]
```diff
@@ -0,0 +1,49 @@
...
++
+[subs="+attributes,+quotes",options="nowrap"]
+----
+$ oc version
+oc v3.11.0+0cbc58b
+----
+
+. Run following commands to create a dummy project to find the URL that this OpenShift instance is using to deploy applications.
```
Comment (rkratky) [inline src/main/pages/che-7/overview/proc_preparing-openshift-3-for-installing-che.adoc]:
```suggestion
. Run the following commands to create a dummy project to find the URL that this OpenShift instance is using to deploy applications.
```

File: src/main/pages/che-7/overview/proc_preparing-openshift-3-for-installing-che.adoc [hunk 26]
```diff
@@ -0,0 +1,49 @@
...
+----
+$ oc new-project hello-world
+$ oc new-app centos/httpd-24-centos7~https://github.com/openshift/httpd-ex
+$ oc expose svc/httpd-ex
+$ oc get route httpd-ex
+NAME     HOST/PORT                                                PATH     SERVICES PORT     TERMINATION WILDCARD
+httpd-ex httpd-ex-hello-world.apps.rhpds311.openshift.opentlc.com httpd-ex          8080-tcp             None
+----
+
+. Extract the domain from `++httpd-ex-hello-world.apps.rhpds311.openshift.opentlc.com++`. It is the part after the first name: `apps.rhpds311.openshift.opentlc.com`. Remember this URL as `<OPENSHIFT_APPS_URL>`.
```
Comment (rkratky) [inline src/main/pages/che-7/overview/proc_preparing-openshift-3-for-installing-che.adoc]:
```suggestion
. Extract the domain from `++httpd-ex-hello-world.apps.rhpds311.openshift.opentlc.com++`. It is the part after the first name: `apps.rhpds311.openshift.opentlc.com`. Remember this URL as `__<OPENSHIFT_APPS_URL>__`.
```

**Review by rkratky — CHANGES_REQUESTED** [review]

**Comment by rkratky** [comment]
@themr0c, I submitted https://github.com/themr0c/che-docs/pull/1 against your topic branch. It should fix all the comments above. Please, let me know if it works for you.

**Comment by themr0c** [comment]
@rkratky I integrated your PR and some other fixes.

**Review by rkratky — APPROVED** [review]

**Pull request merged**